hack/generate-template-variable-doc.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import re
+
+whereami = os.path.abspath(__file__)
+os.chdir(os.path.dirname(whereami))
+
+template_file = open('../eks-worker-al2.json')
+template = json.load(template_file)
+template_file.close()
+
+default_var_file = open('../eks-worker-al2-variables.json')
+default_vars = json.load(default_var_file)
+default_var_file.close()
+
+all_vars = {}
+
+for var in template['variables']:
+  all_vars[var] = None
+for var, default_val in default_vars.items():
+  all_vars[var] = default_val
+
+doc_file_name = '../doc/USER_GUIDE.md'
+doc_file = open(doc_file_name)
+doc = doc_file.read()
+doc_file.close()
```

Comment:
I'm aware it's just `hack`, not a necessary change. I would try to make it more like `python3`

```
template = {}
with open('../eks-worker-al2.json') as f:
    template = json.load(f)

default_vars = {}
with open('../eks-worker-al2-variables.json') as f:
    default_vars = json.load(f)

all_vars = {}

for var in template['variables']:
    all_vars[var] = None

for var, default_val in default_vars.items():
    all_vars[var] = default_val

doc = None
with open('../doc/USER_GUIDE.md') as f:
    doc = f.read()
```

or

```
#!/usr/bin/env python3

import json
import os
import re

script_path = os.path.abspath(__file__)
os.chdir(os.path.dirname(script_path))

TEMPLATE = '../eks-worker-al2.json'
DEFAULT_VARIABLES = '../eks-worker-al2-variables.json'
USER_GUIDE = '../doc/USER_GUIDE.md'

# early exit if any one of file not exist
if False in [os.path.exists(check) for check in [TEMPLATE, DEFAULT_VARIABLES, USER_GUIDE]]:
    os._exit(1)

template = json.load(open(PACKER_TEMPLATE, 'r'))
default_vars = json.load(open(PACKER_DEFAULT_VARIABLES, 'r'))
user_guide = open(USER_GUIDE, 'r').read()

# initialize
all_vars = {}
for var in template['variables']:
    all_vars[var] = None

# override
for var, default_val in default_vars.items():
    all_vars[var] = default_val
```

Reply:
thx